lib/arclight/custom_component.rb

```diff
@@ -28,11 +28,20 @@ def to_solr(solr_doc = {})
       Solrizer.insert_field(solr_doc, 'level', formatted_level, :facetable) # human-readable for facet `level_sim`
       Solrizer.insert_field(solr_doc, 'date_range', formatted_unitdate_for_range, :facetable)
       Solrizer.insert_field(solr_doc, 'access_subjects', access_subjects, :facetable)
+      resolve_repository(solr_doc)
       solr_doc
     end
 
     private
 
+    attr_accessor :repository
+
+    def resolve_repository(solr_doc)
+      @repository ||= solr_doc['repository_ssm']
```

Comment:
I have to admit, I had to read this over a few times to grok what was going on here.  The setting of the accessor that's used in the mixin adds a bit to the cognitive load of this code.

What would you think about passing the value as a parameter to the `repository_as_configured` method?